### PR TITLE
Issue 136.chia

### PR DESCRIPTION
**Pull request by mahchiahui**
# Pull request questions

## Which issue does this address

Issue number: #136 

## Why was change needed

Needed authentication functionality in HTTP API gateway and Websocket API gateway

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,3 +71,48 @@ Detailed instructions at
...
+Set Environment Variables
+```
+export AWS_COGNITO_USERNAME="<replace-with-username>" \
+&& export AWS_COGNITO_PASSWORD="<replace-with-password>" \
+&& export AWS_COGNITO_CLIENT_ID="<replace-with-appclient-id>" \
+&& export AWS_COGNITO_POOL_ID="<replace-with-userpool-id>" \
+&& export API_URL="<replace-with-api-url>" \
+&& export AWS_REGION="<replace-with-aws-region>";
+```
+
```
Comment (docktermj) [inline README.md:86]: On this one,  I think  having individual `exports` might be less confusing.  Example:

```console
export AWS_COGNITO_USERNAME="<replace-with-username>"
export AWS_COGNITO_PASSWORD="<replace-with-password>"
export AWS_COGNITO_CLIENT_ID="<replace-with-appclient-id>"
export AWS_COGNITO_POOL_ID="<replace-with-userpool-id>"
export API_URL="<replace-with-api-url>"
export AWS_REGION="<replace-with-aws-region>"
```

File: cloudformation.yaml [hunk 2]
```diff
@@ -1154,6 +1231,223 @@ Resources:
...
+  HTTPApiGateway:
+    Condition: IfUsingApiServer
+    Type: AWS::ApiGatewayV2::Api
+    Properties:
+      Description: Senzing HTTP API Gateway
+      Name: !Sub "${AWS::StackName}-http-api-gateway"
+      ProtocolType: HTTP
+      
+  WebSocketApiGateway:
+    Condition: IfUsingWeb
```
Comment (roncewind) [inline cloudformation.yaml:1247]: Is this right?  I don't recall the semantics around IfUsingApiServer versus IfUsingWeb.

**Review by roncewind — CHANGES_REQUESTED** [review]
Hey Chia,  the convention that Mike set up in the CFTs is for the resource type to be the last line for the resource.  I think it would be a good idea if we continued to follow that convention.  Thanks!

File: README.md (hunk 1 above)
Comment (mahchiahui) [reply README.md:86]: done

File: cloudformation.yaml (hunk 2 above)
Comment (mahchiahui) [reply cloudformation.yaml:1247]: hmm should i switch all ifusingapiserver to ifusingweb?

I just looked through and ifusingweb encompasses more conditions

```
IfUsingApiServer: !Or
    - !Condition IfRunApiServer
    - !Condition IfRunSwagger
    - !Condition IfRunWebApp
 
IfUsingWeb: !Or
    - !Condition IfRunApiServer
    - !Condition IfRunJupyter
    - !Condition IfRunSwagger
    - !Condition IfRunWebApp
    - !Condition IfRunXterm
```

Comment (roncewind) [reply cloudformation.yaml:1247]: Seems like you only need the API server for the WebSocket interface?  Hm... also of note is that only the POC server has websockets (I believe), so we should probably make a note of that somewhere.  so much to do.  :-)

Comment (mahchiahui) [reply cloudformation.yaml:1247]: hmmm should we then revamp the discussion some of the conditions?

**Review by docktermj — APPROVED** [review]

**Pull request closed without merging**